### PR TITLE
#1194 - Directly predict annotation type

### PR DESCRIPTION
**Pull request by jcklie**
**What's in the PR**
* Monkey patch score feature to every layer's feature
* Monkey patch feature which tells you whether a fs is a prediction or not
* Refactor recommender to use abstract class instead of interface
* Fix some tests
* Recommender directly predict the annotation instead of `PredictedSpan

**How to test manually**
* Test some recommenders in INCEpTION and check whether the score is there

**Automatic testing**
* [x] PR includes unit tests

**Documentation**
* [x] PR updates documentation

### CODE REVIEW

**Comment by jcklie** [comment]
I still need to fix the `PredictionTask` I think and update the documentation.

**Comment by reckart** [comment]
The beta release is done - so feel free to merge things again into master.

**Comment by jcklie** [comment]
Jenkins, can you test this please?

File: inception-imls-external/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/external/RemoteStringMatchingNerRecommender.java [hunk 1]
```diff
@@ -97,21 +101,18 @@ public String predict(String aPredictionRequestJson)
         PredictionRequest request = deserializePredictionRequest(aPredictionRequestJson);
         CAS cas = deserializeCas(request.getDocument().getXmi(), request.getTypeSystem());
 
-        recommendationEngine.predict(context, cas);
+        // Only work on real annotations, not on predictions
+        Type predictedType = CasUtil.getType(cas, recommender.getLayer().getName());
+        Feature feature = predictedType.getFeatureByBaseName("predicted");
```
Comment (reckart) [inline inception-imls-external/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/external/RemoteStringMatchingNerRecommender.java]: Should use a feature name that is unlikely to clash with a user-defined name, e.g. "__predicted__". Also should use a constant for this feature name.

File: inception-recommendation/src/test/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImplIntegrationTest.java [hunk 2]
```diff
@@ -162,20 +193,40 @@ private Project createProject(String aName)
...
         layer.setUiName("test ui name");
         layer.setAnchoringMode(false, false);
        
         return testEntityManager.persist(layer);
     }
 
-    public AnnotationFeature createAnnotationFeature(AnnotationLayer aLayer, String aName)
+    private User createUser()
+    {
+        User user = new User();
```
Comment (reckart) [inline inception-recommendation/src/test/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImplIntegrationTest.java]: `return new User()`?

Comment (reckart) [reply inception-recommendation/src/test/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImplIntegrationTest.java]: Why  a method for this at all?

Comment (jcklie) [reply inception-recommendation/src/test/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImplIntegrationTest.java]: I did it to conform to the other setup functions, if you need to add manipulate the user then its already there for you.

**Comment by jcklie** [comment]
@reckart Can I merge it now?

**Pull request merged**